pkg/istio-agent/xds_proxy_delta.go

```diff
@@ -188,36 +195,37 @@ func (p *XdsProxy) handleUpstreamDeltaResponse(con *ProxyConnection) {
 			// TODO: separate upstream response handling from requests sending, which are both time costly
 			proxyLog.Debugf("response for type url %s", resp.TypeUrl)
 			metrics.XdsProxyResponses.Increment()
-			// TODO(howardjohn) implement handlers. We need to make handlers take in a list of
-			// resources or something probably
-			//if h, f := p.handlers[resp.TypeUrl]; f {
-			//	err := h(resp)
-			//	var errorResp *google_rpc.Status
-			//	if err != nil {
-			//		errorResp = &google_rpc.Status{
-			//			Code:    int32(codes.Internal),
-			//			Message: err.Error(),
-			//		}
-			//	}
-			//	// Send ACK/NACK
-			//	con.requestsChan <- &discovery.DiscoveryRequest{
-			//		VersionInfo:   resp.VersionInfo,
-			//		TypeUrl:       resp.TypeUrl,
-			//		ResponseNonce: resp.Nonce,
-			//		ErrorDetail:   errorResp,
-			//	}
-			//	continue
-			//}
+			if h, f := p.handlers[resp.TypeUrl]; f {
+				if len(resp.Resources) == 0 {
+					// Empty response, nothing to do
+					// This assumes internal types are always singleton
+					return
+				}
+				err := h(resp.Resources[0].Resource)
+				var errorResp *google_rpc.Status
+				if err != nil {
+					errorResp = &google_rpc.Status{
+						Code:    int32(codes.Internal),
+						Message: err.Error(),
+					}
+				}
+				// Send ACK/NACK
+				con.deltaRequestsChan <- &discovery.DeltaDiscoveryRequest{
+					TypeUrl:       resp.TypeUrl,
+					ResponseNonce: resp.Nonce,
+					ErrorDetail:   errorResp,
+				}
+				continue
+			}
 			switch resp.TypeUrl {
-			// TODO: fix WASM
-			// case v3.ExtensionConfigurationType:
-			//	//if features.WasmRemoteLoadConversion {
-			//	//	// If Wasm remote load conversion feature is enabled, rewrite and send.
-			//	//	go p.rewriteAndForward(con, resp)
-			//	//} else {
-			//		// Otherwise, forward ECDS resource update directly to Envoy.
-			//		forwardDeltaToEnvoy(con, resp)
-			//	}
+			case v3.ExtensionConfigurationType:
```

Comment:
why the switch/case here? If we are just checking for `features.WasmRemoteLoadConversion` and that the type is `ExtensionConfigurationType`, couldn't it be:

```go
if resp.TypeUrl == v3.ExtensionConfigurationType && features.WasmRemoteLoadConversion {
  go p.deltaRewriteAndForward(con, resp)
}
forwardDeltaToEnvoy(con, resp)
```

And then remove the `forwardDeltaToEnvoy` in `deltaRewriteAndForward`?

Reply:
yeah that is probably simpler. I was just blindly copying from the non-delta code